### PR TITLE
MM-47284: Migrate /autocomplete/ tests to TypeScript

### PR DESCRIPTION
**Pull request by sinansonmez**
#### Summary
Following files under /integration/autocomplete are converted to Typescript

helpers.js
common_test.js

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/21291
JIRA: https://mattermost.atlassian.net/browse/MM-47284

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: e2e/cypress/tests/integration/autocomplete/helpers.ts [hunk 1]
```diff
@@ -278,7 +287,7 @@ function createChannel(channelType, teamId, userToAdd = null) {
     });
 }
 
-function generatePrefixedUser(user = {}, prefix) {
+function generatePrefixedUser(user: UserInfo, prefix) {
```
Comment (josephbaylon) [inline e2e/cypress/tests/integration/autocomplete/helpers.ts:290]: Can we use existing `UserProfile` type for this?

**Review by josephbaylon — COMMENTED** [review]
Thanks @sinansonmez !! One minor query below

**Review by josephbaylon — APPROVED** [review]
LGTM. Thanks @sinansonmez !!

File: e2e/cypress/tests/integration/autocomplete/helpers.ts [hunk 2]
```diff
@@ -1,6 +1,9 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import {AdminConfig} from '@mattermost/types/lib/config';
+import {UserProfile} from '@mattermost/types/lib/users';
```
Comment (furqanmlk) [inline e2e/cypress/tests/integration/autocomplete/helpers.ts:5]: These types were already under [Cypress namespace](https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/tests/support/index.d.ts#L7-L34)

we can use `Cypress.AdminConfig` etc

**Review by furqanmlk — COMMENTED** [review]
Thanks @sinansonmez 
LGTM!
Just one quick request

File: e2e/cypress/tests/integration/autocomplete/helpers.ts (hunk 2 above)
Comment (sinansonmez) [reply e2e/cypress/tests/integration/autocomplete/helpers.ts]: thanks for the info @furqanmlk I will also use it for other PRs

**Comment by sinansonmez** [comment]
@furqanmlk @josephbaylon When I had a second look, I realized that I could further improve the types. Could you please take a look at my last commit? Sorry for the double work.

**Review by josephbaylon — APPROVED** [review]
@sinansonmez No worries! Thanks for the improvement!

**Review by furqanmlk — APPROVED** [review]
Thanks @sinansonmez 
LGTM !

**Pull request merged**